zookeeper/zookeeper.md

```diff
@@ -0,0 +1,85 @@
+
+### Goal
+
+Provide a centralized service for maintaining configuration information and distributed synchronization. 
+
+### Architecture
+
+Zookeeper architecture please refer to http://zookeeper.apache.org/ and we don't add any patch.
+
+Job status and HA sync info are stored in zookeeper to avoid out-of-sync.
+
+### Dependencies
+
+Cluster-configuration
+
+### Build
+
+* By ` paictl.py `
+
+`
+paictl.py image build -p /path/to/cluster/config -n zookeeper
+`
+
+`
+paictl.py image push -p /path/to/cluster/config -n zookeeper
+`
+
+* Manual build
+
+refer to [dockerfile](https://github.com/Microsoft/pai/blob/master/pai-management/src/zookeeper/dockerfile)
+
+
+### Configuration  
+
+Configuration file is [here](https://github.com/Microsoft/pai/blob/master/pai-management/src/zookeeper/zoo.cfg), you can keep it in most cases.
+
+
+### Reconfiguration
+
+Stop and restart to update configuration. Refresh command in ` paictl.py ` for zookeeper is redundant temporarily.
+
+### Deployment
+
+Deployed by k8s Daemonset.
+
+`
+paictl.py service start -p /path/to/cluster/config -n zookeeper
+`
+
+The command creates k8s Daemonset to deploy service, which select node with zookeeper role to start service.
+
+You can also stop service similarly.
+
+`
+paictl.py service stop -p /path/to/cluster/config -n zookeeper
+`
+
+### Upgrading
+
+Either build new image by yourself or use our [prebuild images](https://hub.docker.com/r/openpai/zookeeper/). For now, upgrade should restart related service.
+
+### Service Metrics
+
+Exported by zookeeper
+
```

Comment:
I remembered ZK has also exported metrics in Prometheus format via rest API, can you check and add it?